pkg/feature/manifest.go

```diff
@@ -7,31 +7,177 @@ import (
 	"html/template"
 	"io"
 	"io/fs"
+	"os"
 	"path"
 	"path/filepath"
+	"regexp"
 	"strings"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+
+	featurev1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/features/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/plugins"
 )
 
 //go:embed templates
 var embeddedFiles embed.FS
 
+const KustomizationFile = "kustomization.yaml"
+
 var (
 	BaseDir        = "templates"
 	ServiceMeshDir = path.Join(BaseDir, "servicemesh")
 	ServerlessDir  = path.Join(BaseDir, "serverless")
 )
 
-type manifest struct {
+// Manifest defines the interface that all manifest types should implement.
+type Manifest interface {
+	Process(data interface{}) ([]*unstructured.Unstructured, error)
+	isPatch() bool
```

Comment:
I am not convinced about this method contract, but I don't have a good solution yet that would be small enough to suggest in this PR... Lets do that in the follow-up.